haystack/nodes/retriever/_embedding_encoder.py

```diff
@@ -41,6 +44,47 @@ def embed_documents(self, docs: List[Document]) -> List[np.ndarray]:
         """
         pass
 
+    def train(
```

Comment:
I think we should make this method abstract and implement it in `_DefaultEmbeddingEncoder` and `_RetribertEmbeddingEncoder` as well. There, we would just raise a `NotImplementedError`, saying that training only works with sentence transformers models. Otherwise, the user would be able to call `train` on FARM-style models, but this call would not have any effect at all without the user knowing it.